chsdi/views/stationboard.py

```diff
@@ -39,68 +40,5 @@ def __init__(self, request):
 
     @view_config(request_method='GET')
     def get_departures(self):
-        current_date = datetime.datetime.now(timezone('Europe/Berlin'))
-        filters = [OevDepartures.stop == self.id, OevDepartures.time > current_date]
-        if self.destination != 'all':
-            filters.append(OevDepartures.destination == self.destination)
-
-        def serialize(t):
-            return t.strftime('%H:%M')
-
-        def serializeDate(t):
-            return t.strftime('%d/%m/%Y %H:%M')
-
-        query = self.request.db.query(OevDepartures).filter(and_(*filters)).order_by(
-            OevDepartures.time
-        ).limit(self.limit)
-
-        try:
-            # Not all the stations are in the table Haltestelle
-            results = [{
-                'id': q.stop,
-                'time': serialize(q.time),
-                'label': q.label,
-                'via': q.via,
-                'currentDate': serializeDate(current_date),
-                'departureDate': serializeDate(q.time),
-                'destinationName': q.haltestelle.name,
-                'destinationId': q.haltestelle.id
-            } for q in query if hasattr(q.haltestelle, 'id')]
-        except OperationalError as e:  # pragma: no cover
-            raise HTTPInternalServerError(e)
-
-        if not results:
-            return [{'destination': 'nodata'}]
-        return results
-
-
-@view_defaults(renderer='jsonp', route_name='stationboard_destination')
-class DestinationView(object):
-
-    def __init__(self, request):
-        self.request = request
-        if request.matched_route.name == 'stationboard_destination':
-            id = request.matchdict['id']
-            if id.isdigit() is False:
-                raise HTTPBadRequest('The id must be an integer.')
-            else:
-                self.id = int(id)
-
-    @view_config(request_method='GET')
-    def get_destination(self):
-        query = self.request.db.query(OevDepartures).filter(
-            OevDepartures.stop == self.id
-        ).distinct(OevDepartures.destination)
-
-        try:
-            # Not all the stations are in the table Haltestelle
-            results = [{
-                'id': q.haltestelle.id,
-                'name': q.haltestelle.name
-            } for q in query if hasattr(q.haltestelle, 'id')]
-        except OperationalError as e:  # pragma: no cover
-            raise HTTPInternalServerError(e)
-
-        if not results:
-            results = [{'destination': 'nodata'}]
+        results = self.ot_api.get_departures(self.id, self.limit)
```

Comment:
Try catch the exceptions you raise in OpenTrans API, raise HTTP Errors accordingly.